sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -707,13 +707,26 @@ size_t coap_put_block1_ok(uint8_t *pkt_pos, coap_block1_t *block1, uint16_t last
     }
 }
 
-size_t coap_opt_put_block2(uint8_t *buf, uint16_t lastonum, coap_block_slicer_t *slicer, bool more)
+size_t coap_opt_put_block(uint8_t *buf, uint16_t lastonum, coap_block_slicer_t *slicer,
+                          bool more, uint16_t option)
 {
-    unsigned szx = _size2szx(slicer->end - slicer->start);
-    unsigned blknum = _slicer_blknum(slicer);
+    coap_block1_t block;
+
+    coap_block_object_init(&block, _slicer_blknum(slicer),
```

Comment:
Here `coap_block_object_init` receives a `bool more` but in it declaration its type is `int`. Shouldn't it be a bool in both cases? Change the documentation accordingly if its the case.

Reply:
Thanks for reading closely. We specifically use coap_opt_put_block() when writing a block, so it's only useful for its `more` parameter to be a bool. At the same time, coap_block_object_init() directly sets the struct attribute values, and the `more` attribute can take on a value of -1, so it makes sense for the function to support that. I think this is a case where the declaration of  coap_opt_put_block() and coap_block_object_init() make sense as they are, and it's OK to let the definition of `bool` as 0 or 1 bridge the gap.

However, the doc comment for coap_block_object_init() only explicitly refers to use of 0 if we don't know if there are more blocks. I will extend the comment to be more precise about use of 1 and 0. However, I'd like to leave the use of -1 as unspecified because we don't expect anyone to use it for this purpose. That may change in the future, and if it does we won't need to change the API to accommodate it.

Reply:
OK!